src/gurt/telemetry.c

```diff
@@ -2095,7 +2095,7 @@ d_tm_get_gauge(uint64_t *val, struct d_tm_stats_t *stats, uint64_t *shmem_root,
 		if (node->dtn_protect)
 			D_MUTEX_LOCK(&node->dtn_lock);
 		*val = metric_data->dtm_data.value;
-		if (stats != NULL) {
+		if ((stats != NULL) && (dtm_stats != NULL)) {
```

Comment:
Is this the right thing to do if dtm_stats == NULL? Basically I'm wondering if we should just exit early with an error in that case.

Reply:
The check for dtm_stats != NULL is for a Coverity fix.  It thinks it might be dereferenced even if it's a null pointer.  That's one of those 'cannot happen' problems by design, but their scan does not know that.   Creating the stats is not optional for metrics that support them.  And, failing to allocate that memory for the stats is handled as an error.  Because that particular error is already handled, it isn't really necessary to handle it again here.  It would be a code path that is never executed.

Reply:
Ah, I see. If it's not possible to return NULL, a `D_ASSERT(dtm_stats != NULL);` after the `d_tm_conv_ptr` call would be appropriate.